docker/tools/environment.sh

```diff
@@ -55,7 +55,7 @@ fi
 
 # this variable is needed for the build for signing the
 # eservice and pservice enclaves
-export PDO_ENCLAVE_CODE_SIGN_PEM=${PDO_SGX_KEY_ROOT}/enclave_code_sign.pem
+export PDO_ENCLAVE_CODE_SIGN_PEM=/tmp/enclave_code_sign.pem
 
```

Comment:
why are we putting this in /tmp? how about $PDO_HOME/keys/sgx?

Reply:
why not `$PDO_HOME/docker/xfer/services/keys/sgx` where also other sgx related key material has to be for stuff, e.g., docker to work?
BTW: we don't document that for docker we have to put stuff there (because that directory happens to be  `$PDO_SGX_KEY_ROOT` in docker? (currently our docu only tells you in `docs/install.md` what `$PDO_SGX_KEY_ROOT` has to contain but the directories which that variable points to in that documentation doesn't help you for docker.

Reply:
> why not $PDO_HOME/docker/xfer/services/keys/sgx where also other sgx related key material has to be for stuff, e.g., docker to work?

Because the xfer is meant to be mounted, but the enclave code signing key is actually a build time artifact.
AFAIK, it's not possible to mount volumes at build time.

Reply:
> > why not $PDO_HOME/docker/xfer/services/keys/sgx where also other sgx related key material has to be for stuff, e.g., docker to work?
> 
> Because the xfer is meant to be mounted, but the enclave code signing key is actually a build time artifact. AFAIK, it's not possible to mount volumes at build time.

Good point that it is used inside docker. There is also no mount during build as you say (you only can COPY).  So the problem really is that we use that script both at build and at runtime and so just using xfer does not work for build?  But we could (optionally) have a key in xfer and copy it during build to the "right" place?  Anyway, no strong feeling, i guess before "real" code signing keys really would matter there is a whole lot of other stuff we would have to do first to make it production-ready secure, e.g., testing that we are not in debug mode :-)

Reply:
Yes, we could generate the key outside and then inject it in the same place.
I'll leave as is for now to reduce complexity of PR.
However, I think it's desirable to go that route because we could probably remove the other `build/keys/sgx_mode_*` folders.